packages/block-library/src/term-template/variations.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { list, grid } from '@wordpress/icons';
+
+const variations = [
+	{
+		name: 'list',
+		title: __( 'List' ),
+		description: __( 'Display terms in a list layout.' ),
+		attributes: { layout: { type: 'default' } },
+		isDefault: true,
+		icon: list,
+		scope: [ 'block', 'inserter' ],
+		innerBlocks: [
+			[
+				'core/group',
+				{
+					layout: {
+						type: 'flex',
+						orientation: 'horizontal',
+					},
+					style: {
+						spacing: {
+							blockGap: '0.5rem',
+						},
+					},
+					metadata: {
+						name: __( 'Term Name with Count' ),
+					},
+				},
+				[
+					[
+						'core/paragraph',
+						{
+							metadata: {
+								name: __( 'Term Name' ),
+								bindings: {
+									content: {
+										source: 'core/term-data',
+										args: {
+											key: 'name',
+										},
+									},
+								},
+							},
+						},
+					],
+					[
+						'core/paragraph',
+						{
+							placeholder: __( '(count)' ),
+							metadata: {
+								name: __( 'Term Count' ),
+								bindings: {
+									content: {
+										source: 'core/term-data',
+										args: {
+											key: 'count',
+										},
+									},
+								},
+							},
+						},
+					],
+				],
+			],
+		],
+	},
+	{
+		name: 'grid',
+		title: __( 'Grid' ),
+		description: __( 'Display terms in a grid layout.' ),
+		attributes: { layout: { type: 'grid', columnCount: 3 } },
+		icon: grid,
+		scope: [ 'block', 'inserter' ],
+		innerBlocks: [
```

Comment:
Minor but `innerBlocks` are the same in both variations so you can consider extracting them to a variable. Obviously, it's fine keeping them as-is so it's more declarative and easier to modify in a future.